web/shapeworks/src/api/rest.ts

```diff
@@ -26,8 +27,24 @@ export async function getDataObjectsForSubject(subjectId: number): Promise<DataO
             params: {subject: subjectId}
         })
     }))).map((response, index) => {
-        return response.data.results.map((result: DataObject) => Object.assign(
-            result, {type: dataTypes[index]}
-        ))
+        return response.data.results.map((result: DataObject) => {
+            const type = dataTypes[index]
+            if(type !== 'image'){
+                // don't await this, let particles load in after
+                getOptimizedParticlesForDataObject(type, result.id)
+            }
+            return Object.assign(result, {type})
+        })
     }).flat(2)
 }
+
+export async function getOptimizedParticlesForDataObject(type: string, id: number){
```

Comment:
I think this makes more sense split into a `rest` method and a `store` method. IMHO api client methods shouldn't modify state, that should be the purview of methods defined in the store.

Reply:
Good point, changed in [348c956](https://github.com/girder/shapeworks-cloud/pull/155/commits/348c956db3815cf93925f612bc93c03b5d5955d3)